web/src/components/redactors/EditRedactor.tsx

```diff
@@ -0,0 +1,433 @@
+import React, { useEffect, useReducer } from "react";
+import { Link, useHistory, useRouteMatch } from "react-router-dom";
+import { KotsPageTitle } from "@components/Head";
+import AceEditor, { Marker } from "react-ace";
+import "brace/mode/text";
+import "brace/mode/yaml";
+import "brace/theme/chrome";
+
+import Loader from "../shared/Loader";
+import { Utilities } from "../../utilities/utilities";
+
+import "../../scss/components/redactors/EditRedactor.scss";
+import Icon from "../Icon";
+
+import { KotsParams } from "@types";
+import { useSelectedApp } from "@features/App";
+
+type State = {
+  activeMarkers?: Marker[];
+  createConfirm: boolean;
+  createErrMsg: string;
+  creatingRedactor: boolean;
+  editConfirm: boolean;
+  editingErrMsg: string;
+  editingRedactor: boolean;
+  isLoadingRedactor: boolean;
+  redactorEnabled: boolean;
+  redactorErrMsg: string;
+  redactorName: string;
+  redactorYaml: string;
+};
+
+// TODO: upgrade AceEditor so we can use the type definitions
+let aceEditor: AceEditor | null;
```

Comment:
this thing is 4 major versions behind. This was originally stored in the instance but storing here for now. I think this should be fine unless we have multiple editors visible at once. Not even sure why this is needed and maybe we can get rid of it after upgrading `AceEditor`